### PR TITLE
Applicants user profile

### PR DESCRIPTION
**Pull request by mauditecandela**
I'm creating a PR to my fork in order to deploy it to staging and show it to the Travis CI Foundation.

### CODE REVIEW

File: app/assets/stylesheets/form.scss [hunk 1]
```diff
@@ -52,6 +48,7 @@ textarea {
   padding: .3em;
   border: 1px solid $border-grey;
   border-radius: 2px;
+  margin-bottom: 2em;
```
Comment (guainumbi) [inline app/assets/stylesheets/form.scss:51]: accidentally changed margin-bottom from 1em to 2em. I think we should keep 1em instead.

File: app/assets/stylesheets/form.scss [hunk 2]
```diff
@@ -168,3 +165,11 @@ input.has-dependent:checked + div.form_field {
     white-space: nowrap;
   }
 }
+
+.user_information {
```
Comment (guainumbi) [inline app/assets/stylesheets/form.scss:169]: we dont need this class

Comment (mauditecandela) [inline app/assets/stylesheets/form.scss:169]: No longer needed. 😀

File: app/controllers/applications_controller.rb [hunk 3]
```diff
@@ -38,11 +39,25 @@ def destroy
...
     end
+
+    def ensure_correct_user
+      @applicant = User.find_by(id: Application.find(params[:id]).applicant_id)
+      unless @applicant == current_user || admin_user?
+        redirect_to root_path
+      end
+    end
+
+    def check_signed_user
```
Comment (guainumbi) [inline app/controllers/applications_controller.rb:58]: isnt this the same method as "ensure_correct_user"?

Comment (guainumbi) [reply app/controllers/applications_controller.rb:58]: (at least same functionality)

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -29,6 +48,18 @@ def ensure_correct_user
     end
 
     def user_params
-      params.require(:user).permit(:email, :password)
+      params.require(:user).permit(:name, :email, :password)
+    end
+
+    def user_from_params
```
Comment (guainumbi) [inline app/controllers/users_controller.rb:54]: cosmetics: sort variables in order of name - email - password

File: app/views/applications/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    <% if current_user %>
+      <div class="form_field">
+        <%= f.form_field :text_field, :name, 'Name', value: current_user.name %>
+        <%= f.form_field :text_field, :email, 'Email', value: current_user.email %>
+        <%= f.form_field :text_field, :email_confirmation, 'Email Confirmation' %>
+      </div>
+    <% else %>
+      <%= f.form_field :text_field, :name, 'Name' %>
+      <%= f.form_field :text_field, :email, 'Email' %>
+      <%= f.form_field :text_field, :email_confirmation, 'Email Confirmation' %>
```
Comment (guainumbi) [inline app/views/applications/_form.html.erb:27]: we can take out one line "<%= f.form_field :text_field, :email_confirmation, 'Email Confirmation' %>" and place the other below <% end %>

File: db/schema.rb [hunk 6]
```diff
@@ -11,95 +10,93 @@
...
+    t.string "remember_token", limit: 128, null: false
+    t.boolean "admin", default: false, null: false
+    t.string "name"
+    t.index ["email"], name: "index_users_on_email"
+    t.index ["remember_token"], name: "index_users_on_remember_token"
   end
 
-  add_index "users", ["email"], name: "index_users_on_email", using: :btree
-  add_index "users", ["remember_token"], name: "index_users_on_remember_token", using: :btree
-
```
Comment (guainumbi) [inline db/schema.rb]: why delete using:btree? was this intentional?

File: test/integration/new_application_test.rb [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    visit event_path(@event.id)
+
+    click_button "Apply"
+
+    click_button "Submit Application"
+
+    assert page.text.include?("5 errors stopped this application from being saved")
+  end
+
+  test 'keeps information introduced in the application event if the user is signed in' do
```
Comment (mauditecandela) [inline test/integration/new_application_test.rb:56]: "manually introduced"

Comment (mauditecandela) [reply test/integration/new_application_test.rb:56]: and IN the application (does not change users profile) -- maybe we should also check that the user name is not changed

File: db/schema.rb [hunk 8]
```diff
@@ -11,95 +10,93 @@
...
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.integer "event_id"
+    t.text "attendee_info_1"
+    t.text "attendee_info_2"
+    t.boolean "ticket_needed", default: false, null: false
+    t.boolean "travel_needed", default: false, null: false
+    t.boolean "accommodation_needed", default: false, null: false
+    t.boolean "visa_needed", default: false, null: false
+    t.string "application"
```
Comment (guainumbi) [inline db/schema.rb]: new attribute

File: test/integration/new_application_test.rb [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    page.fill_in 'application_attendee_info_1', with: 'I want to learn how to code'
+    page.fill_in 'application_attendee_info_2', with: 'I am an underrepresented minority'
+    page.fill_in 'application_email_confirmation', with: @user.email
+    page.check 'application[terms_and_conditions]'
+
+    click_button "Submit Application"
+
+    assert page.text.include?("You have successfully applied for #{@event.name}")
+
+    assert_equal Application.find_by(applicant_id: @user.id).name, "New name"
```
Comment (mauditecandela) [inline test/integration/new_application_test.rb:73]: assert_equal @user.name, "Awesome name"

File: test/integration/application_page_test.rb [hunk 10]
```diff
@@ -0,0 +1,59 @@
+require 'test_helper'
+
```
Comment (guainumbi) [inline test/integration/application_page_test.rb:2]: needs refactoring

File: test/integration/new_application_test.rb [hunk 11]
```diff
@@ -0,0 +1,75 @@
+require 'test_helper'
+
+feature 'New Application' do
+  def setup
+    @admin = make_admin
+    @user = make_user
+    @event = make_event(organizer_id: @admin.id, approved: true)
+  end
+
+  test 'creates an application after an unlogged user introduced their information' do
```
Comment (guainumbi) [inline test/integration/new_application_test.rb:10]: "creates an application for logged-out user" ?

File: test/integration/new_application_test.rb (hunk 7 above)
Comment (guainumbi) [inline test/integration/new_application_test.rb:56]: this sentence is a bit unclear to me

File: test/integration/user_settings_page_test.rb [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+    click_link 'Settings'
+
+    assert page.text.include?('Edit your credentials')
+    page.must_have_selector("form input[name='user[name]']")
+    page.fill_in 'user_name', with: 'My Name'
+    page.fill_in 'user_password', with: @user.password
+
+    click_button 'Update User'
+
+    assert page.text.include?('You have successfully updated your user data.')
```
Comment (mauditecandela) [inline test/integration/user_settings_page_test.rb:22]: assert_equal @user.name, "My Name"

File: test/integration/user_applications_page_test.rb [hunk 13]
```diff
@@ -0,0 +1,24 @@
+require 'test_helper'
+
+feature 'User Applications Page' do
+  def setup
+    @user = make_user
+    @event = make_event(name: 'Applicants event')
+    @application = make_application(@event, applicant_id: @user.id)
+  end
+
+  test 'shows a section to Your Applications where the users applications are displayed' do
```
Comment (guainumbi) [inline test/integration/user_applications_page_test.rb:10]: minor: "...section **in** Your Applications..." ;)

File: test/test_helper.rb [hunk 14]
```diff
@@ -21,6 +21,7 @@ class ActiveSupport::TestCase
 
   def make_user(user_params = {})
     defaults = {
+      name: 'Awesome name',
```
Comment (guainumbi) [inline test/test_helper.rb:24]: nice! :)

File: app/controllers/applications_controller.rb (hunk 3 above)
Comment (mauditecandela) [reply app/controllers/applications_controller.rb:58]: as discussed we'll change this method to "set_applicant_id"

File: app/views/layouts/_header.html.erb [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+                <li class="dropdown-item"><%= navbar_link_to 'Tags', tags_path %></li>
+              <% end %>
+              <li class="dropdown-item"><%= navbar_link_to 'Settings', edit_user_path(current_user) %></li>
+              <li class="dropdown-item"><%= navbar_link_to 'Your Events', user_path(current_user) %></li>
+              <li class="dropdown-item"><%= navbar_link_to 'Your Applications', user_applications_path(current_user) %></li>
+              <li class="dropdown-item"><%= link_to 'Sign out', sign_out_path, method: :delete %></li>
+            </div>
+          </div>
+        <% else %>
+          <li><%= link_to 'Sign in', sign_in_path %></li>
```
Comment (mauditecandela) [inline app/views/layouts/_header.html.erb:32]: this could be also alligned to the right, for consistency with the dropdown menu

**Pull request merged**